### DISCUSSION TITLE
UWP to Uno : Tasks and Backgroundworker

### DISCUSSION THREAD

**Discussion by pkarsten** [unoplatform/uno#16339] [Q&A]
Hello, I have a UWP APP which is currently running on a Raspberry Pi (3B+) with Windows IOT. 

I would like to move the APP to a RaspberryPi with Linux. 

The APP basically consists of a page that displays a background image, with a calendar/to-do displayed above it. 

I get the data from OneDrive using the MS Graph API. 

The app starts with DashBoard.xaml, and is bound to a ViewModel, in the "OnLoad" of the ViewModel, I register a BackgroundTaskRegistration. 

The task gets an image from the MS Graph API and updates the UI. 

On the Linux Raspberry I probably can't use these tasks, because they are pure Windows tasks? I'm having a bit of a hard time finding a way to reprogram my app. 

What options do I have for Linuux Raspberry (is it the SKIA.GTK?). 

The APP should do the following, a screen, in the background a picture is loaded, when the download is finished, it should be displayed, another task loads calendar entries, when finished, these should be displayed. 

I'm having a hard time figuring out which methods I can use for the Raspberry and what I need

Here is my Current UWP APp: 
https://github.com/pkarsten/HelloWindowsIotGit

Thanks.

**Comment by jeromelaban** [comment]
Background tasks on Linux are handling like any other threads, you do not need to use background workers (in the UWP sense). You can create a long-running thread or Task at the start of your application, then run operations at predefined times or using separate triggers.
